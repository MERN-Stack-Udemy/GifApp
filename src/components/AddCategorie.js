import React, { useState } from "react";
import PropTypes from 'prop-types'


export const AddCategorie = ( {setCategories} ) => {

  const [inputvalue, setInputvalue] = useState(''); // ''

  const handleChange = (e) => {
    setInputvalue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if(inputvalue.trim().length > 2){
      setCategories(categories => [...categories, inputvalue])
      setInputvalue( "" );
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input 
        type = "text" 
        value = {inputvalue} 
        onChange = {handleChange} />
    </form>
  );

};

AddCategorie.propTypes = {
  setCategories : PropTypes.func.isRequired,
}
