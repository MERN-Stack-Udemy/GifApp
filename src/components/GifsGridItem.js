import React from 'react'

export const GifsGridItem = ( {image, title} ) => {
  
  return (
    <div className="card">
      <img 
        src={image.url}
        alt= {title}
      />
      <p>{title}</p>
    </div>
  )
}
