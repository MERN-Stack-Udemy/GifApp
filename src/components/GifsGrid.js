import React, { useEffect, useState } from 'react'
import { getGifs } from '../Helpers/getGifs';
import { GifsGridItem } from './GifsGridItem';

export const GifsGrid = ({ categorie }) => {

  const [images, setImages] = useState([]);

  useEffect(() => {

    getGifs(categorie)
    .then( setImages )

  }, [categorie] );

 


  return (
    <>
      <h3>{categorie}</h3>
      <div className="card-grid">
        {images.map( (img) => (
          <GifsGridItem 
            key = {img.id} 
            {...img}
          />
        ))}
        
      </div>
    </>
  )
}
