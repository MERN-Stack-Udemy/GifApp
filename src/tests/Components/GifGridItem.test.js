import { shallow } from "enzyme"
import { GifsGridItem } from "../../components/GifsGridItem"


describe('Test about GifsGridItem', () => {
  
  const url = 'https://api.giphy.com/seach?rick+y+morty.jpg';
  const title = 'rick y morty';
  const wrapper = shallow(<GifsGridItem  url={url} title={title} />);
  
  test('test snapshot of GifsGritItem', () => {
    expect(wrapper).toMatchSnapshot();
  })
  
  test('check title of <p/> text()', () => {
    const p_title = wrapper.find('p').text();
    expect( p_title ).toBe( title );
  })
  
  test('<img/> must contain the attributes: \'alt\' and \'src\'', () => {
    const img = wrapper.find('img');
    expect( img.prop('src') ).toBe( url );
    expect( img.prop('alt') ).toBe( title );
  })
  
  test('<div/> should contains the className: \'animate__bounceIn\' ', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    expect( className.includes('animate__bounceIn') ).toBe(true)
  })
  

})
