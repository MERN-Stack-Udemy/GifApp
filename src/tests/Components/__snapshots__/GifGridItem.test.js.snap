// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test about GifsGridItem test snapshot of GifsGritItem 1`] = `
<div
  className="card animate__bounceIn"
>
  <img
    alt="rick y morty"
    src="https://api.giphy.com/seach?rick+y+morty.jpg"
  />
  <p>
    rick y morty
  </p>
</div>
`;
